src/transformers/models/bigscience176b/modeling_bigscience176b.py

```diff
@@ -680,24 +809,24 @@ def forward(
             position_ids = position_ids.unsqueeze(0).view(-1, input_shape[-1])
 
         # BigScience176BAttention mask.
-        if attention_mask is not None:
-            if batch_size <= 0:
-                raise ValueError("batch_size has to be defined and > 0")
-            attention_mask = attention_mask.view(batch_size, -1)
-            # We create a 3D attention mask from a 2D tensor mask.
-            # Sizes are [batch_size, 1, 1, to_seq_length]
-            # So we can broadcast to [batch_size, num_heads, from_seq_length, to_seq_length]
-            # this attention mask is more simple than the triangular masking of causal attention
-            # used in OpenAI GPT, we just need to prepare the broadcast dimension here.
-            attention_mask = attention_mask[:, None, None, :]
-
-            # Since attention_mask is 1.0 for positions we want to attend and 0.0 for
-            # masked positions, this operation will create a tensor which is 0.0 for
-            # positions we want to attend and -10000.0 for masked positions.
-            # Since we are adding it to the raw scores before the softmax, this is
-            # effectively the same as removing these entirely.
-            attention_mask = attention_mask.to(dtype=self.dtype)  # fp16 compatibility
-            attention_mask = (1.0 - attention_mask) * -10000.0
+        # if attention_mask is not None:
+        #     if batch_size <= 0:
+        #         raise ValueError("batch_size has to be defined and > 0")
+        #     attention_mask = attention_mask.view(batch_size, -1)
+        #     # We create a 3D attention mask from a 2D tensor mask.
+        #     # Sizes are [batch_size, 1, 1, to_seq_length]
+        #     # So we can broadcast to [batch_size, num_heads, from_seq_length, to_seq_length]
+        #     # this attention mask is more simple than the triangular masking of causal attention
+        #     # used in OpenAI GPT, we just need to prepare the broadcast dimension here.
+        #     attention_mask = attention_mask[:, None, None, :]
+
+        #     # Since attention_mask is 1.0 for positions we want to attend and 0.0 for
+        #     # masked positions, this operation will create a tensor which is 0.0 for
+        #     # positions we want to attend and -10000.0 for masked positions.
+        #     # Since we are adding it to the raw scores before the softmax, this is
+        #     # effectively the same as removing these entirely.
+        #     attention_mask = attention_mask.to(dtype=self.dtype)  # fp16 compatibility
+        #     attention_mask = (1.0 - attention_mask) * -10000.0
```

Comment:
We'll probably need to pass the attention mask at some point but we can see that later when fixing all the integration tests

Reply:
Actually I am passing the attention masks in the test (since it is done also on the Megatron-LM script I was using, I wanted to ensure the reproducibility using the exact same input, including the masks). I commented this block because this operation is done automatically inside the sclaed masked softmax function by [this function](https://github.com/younesbelkada/transformers/blob/93d270a906a54f3f584d6cbe84196c7088069fa0/src/transformers/models/bigscience176b/modeling_bigscience176b.py#L81) which is [initialized](https://github.com/younesbelkada/transformers/blob/93d270a906a54f3f584d6cbe84196c7088069fa0/src/transformers/models/bigscience176b/modeling_bigscience176b.py#L134) and used by the scaled sofmax function [here](https://github.com/younesbelkada/transformers/blob/93d270a906a54f3f584d6cbe84196c7088069fa0/src/transformers/models/bigscience176b/scale_mask_softmax.py#L74)

Reply:
Since the script has changed a bit (Removed Scaled Mask softmax), I adapted everything on the new commit!